### PR TITLE
Team API error and page

### PR DESCRIPTION
**Pull request by mihaic195**
- [x] Fix crash when user is unauthenticated and tries to navigate to `settings/team`
- [x] Fix 500 when creating a team.

### CODE REVIEW

File: pages/settings/teams.tsx [hunk 1]
```diff
@@ -153,4 +156,15 @@ export default function Teams(props) {
       </SettingsShell>
     </Shell>
   );
-}
\ No newline at end of file
+}
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
```
Comment (emrysal) [inline pages/settings/teams.tsx:161]: I don't think this is the way to fix this issue; rather, when a 401 is thrown during `loadData` we should catch that 401 and redirect to Login instead.

Comment (mihaic195) [reply pages/settings/teams.tsx]: @emrysal Thanks for that. I've updated the code, should be ok now 👍

File: pages/api/teams.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       }
     });
 
-    return res.status(201).setHeader('Location', process.env.BASE_URL + '/api/teams/1').send(null);
+    return res.status(201).json({ message: 'Team created' });
   }
 
-  res.status(404).send(null);
+  res.status(404).json({ message: 'Team not found' });
```
Comment (baileypumfleet) [inline pages/api/teams.ts:36]: Codacy has a fix for the issue: [Replace `'Team·not·found'` with `"Team·not·found"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  res.status(404).json({ message: "Team not found" });
```

File: pages/settings/teams.tsx [hunk 3]
```diff
@@ -1,46 +1,57 @@
...
-    }
-  );
+    })
+    .catch(console.log);
+  }
 
-  useEffect(() => { loadTeams(); }, []);
+  useEffect(() => {
+    loadData();
+  }, []);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:38]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'loadData'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  }, [loadData]);
```

File: pages/settings/teams.tsx [hunk 4]
```diff
@@ -1,46 +1,57 @@
...
+  const handleErrors = async (resp) => {
+    if (!resp.ok) {
+      const err = await resp.json();
+      throw new Error(err.message);
+    }
+    return resp.json();
+  }
+
+  const loadData = () => {
+    fetch('/api/user/membership')
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:27]: Codacy has a fix for the issue: [Replace `'/api/user/membership'` with `"/api/user/membership"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    fetch("/api/user/membership")
```

File: pages/settings/teams.tsx [hunk 5]
```diff
@@ -1,46 +1,57 @@
...
+    .then(handleErrors)
+    .then((data) => {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:29]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then((data) => {
```

File: pages/settings/teams.tsx [hunk 6]
```diff
@@ -86,14 +97,14 @@ export default function Teams(props) {
...
-                <TeamList teams={teams} onChange={loadTeams}>
+                <TeamList teams={teams} onChange={loadData}>
                 </TeamList>
               }
 
               {!!invites.length && <div>
                 <h2 className="text-lg leading-6 font-medium text-gray-900">Open Invitations</h2>
                 <ul className="border px-2 rounded mt-2 mb-2 divide-y divide-gray-200">
-                  {invites.map((team) => <TeamListItem onChange={loadTeams} key={team.id} team={team}></TeamListItem>)}
+                  {invites.map((team) => <TeamListItem onChange={loadData} key={team.id} team={team}></TeamListItem>)}
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:107]: Codacy has a fix for the issue: [Replace `{invites.map((team)·=>·<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>` with `··{invites.map((team)·=>·(⏎······················<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>⏎····················)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
                    {invites.map((team) => (
```

File: pages/settings/teams.tsx [hunk 7]
```diff
@@ -1,46 +1,57 @@
...
+  const loadData = () => {
+    fetch('/api/user/membership')
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
+    .catch(console.log);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:33]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .catch(console.log);
```

File: pages/settings/teams.tsx [hunk 8]
```diff
@@ -1,46 +1,57 @@
 import Head from 'next/head';
-import prisma from '../../lib/prisma';
-import Modal from '../../components/Modal';
 import Shell from '../../components/Shell';
 import SettingsShell from '../../components/Settings';
 import { useEffect, useState } from 'react';
-import { useSession, getSession } from 'next-auth/client';
+import { useSession } from 'next-auth/client';
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:5]: Codacy has a fix for the issue: [Replace `'next-auth/client'` with `"next-auth/client"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
import { useSession } from "next-auth/client";
```

File: pages/settings/teams.tsx [hunk 9]
```diff
@@ -1,46 +1,57 @@
...
+
+  const loadData = () => {
+    fetch('/api/user/membership')
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:32]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      })
```

File: pages/settings/teams.tsx (hunk 8 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:5]: Codacy has a fix for the issue: [Replace `'next-auth/client'` with `"next-auth/client"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
import { useSession } from "next-auth/client";
```

File: pages/settings/teams.tsx (hunk 6 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:107]: Codacy has a fix for the issue: [Replace `{invites.map((team)·=>·<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>` with `··{invites.map((team)·=>·(⏎······················<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>⏎····················)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
                    {invites.map((team) => (
```

File: pages/settings/teams.tsx [hunk 10]
```diff
@@ -1,46 +1,57 @@
...
+
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:32]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      })
```

File: pages/api/teams.ts (hunk 2 above)
Comment (baileypumfleet) [inline pages/api/teams.ts:36]: Codacy has a fix for the issue: [Replace `'Team·not·found'` with `"Team·not·found"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  res.status(404).json({ message: "Team not found" });
```

File: pages/settings/teams.tsx [hunk 11]
```diff
@@ -1,46 +1,57 @@
...
+    .catch(console.log);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:33]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .catch(console.log);
```

File: pages/settings/teams.tsx [hunk 12]
```diff
@@ -1,46 +1,59 @@
...
+  }
 
-  useEffect(() => { loadTeams(); }, []);
+  useEffect(() => {
+    loadData();
+  }, []);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:40]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'loadData'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  }, [loadData]);
```

File: pages/settings/teams.tsx [hunk 13]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{session: Session | null}> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
+      return { redirect: { permanent: false, destination: '/auth/login' } };
+    }
+
+    return {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:178]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  return {
```

File: pages/settings/teams.tsx [hunk 14]
```diff
@@ -1,46 +1,59 @@
+import { GetServerSideProps } from "next"
 import Head from 'next/head';
-import prisma from '../../lib/prisma';
-import Modal from '../../components/Modal';
 import Shell from '../../components/Shell';
 import SettingsShell from '../../components/Settings';
 import { useEffect, useState } from 'react';
+import type { Session } from "next-auth"
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:6]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
import type { Session } from "next-auth";
```

File: pages/api/teams.ts (hunk 2 above)
Comment (baileypumfleet) [inline pages/api/teams.ts:36]: Codacy has a fix for the issue: [Replace `'Team·not·found'` with `"Team·not·found"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  res.status(404).json({ message: "Team not found" });
```

File: pages/settings/teams.tsx [hunk 15]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
     </Shell>
   );
-}
\ No newline at end of file
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{session: Session | null}> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:174]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  if (!session) {
```

File: pages/settings/teams.tsx [hunk 16]
```diff
@@ -1,46 +1,59 @@
+import { GetServerSideProps } from "next"
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:1]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
import { GetServerSideProps } from "next";
```

File: pages/settings/teams.tsx [hunk 17]
```diff
@@ -86,14 +99,14 @@ export default function Teams(props) {
...
-                <TeamList teams={teams} onChange={loadTeams}>
+                <TeamList teams={teams} onChange={loadData}>
                 </TeamList>
               }
 
               {!!invites.length && <div>
                 <h2 className="text-lg leading-6 font-medium text-gray-900">Open Invitations</h2>
                 <ul className="border px-2 rounded mt-2 mb-2 divide-y divide-gray-200">
-                  {invites.map((team) => <TeamListItem onChange={loadTeams} key={team.id} team={team}></TeamListItem>)}
+                  {invites.map((team) => <TeamListItem onChange={loadData} key={team.id} team={team}></TeamListItem>)}
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:109]: Codacy has a fix for the issue: [Replace `{invites.map((team)·=>·<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>` with `··{invites.map((team)·=>·(⏎······················<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>⏎····················)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
                    {invites.map((team) => (
```

File: pages/settings/teams.tsx [hunk 18]
```diff
@@ -1,46 +1,59 @@
...
+    if (!resp.ok) {
+      const err = await resp.json();
+      throw new Error(err.message);
+    }
+    return resp.json();
+  }
+
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:30]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then(handleErrors)
```

File: pages/_app.tsx [hunk 19]
```diff
@@ -1,8 +1,9 @@
 import '../styles/globals.css';
 import {createTelemetryClient, TelemetryProvider} from '../lib/telemetry';
 import { Provider } from 'next-auth/client';
+import type { AppProps } from "next/app"
```
Comment (baileypumfleet) [inline pages/_app.tsx:4]: Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
import type { AppProps } from "next/app";
```

File: pages/settings/teams.tsx [hunk 20]
```diff
@@ -1,46 +1,59 @@
...
+    .then((data) => {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:31]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then((data) => {
```

File: pages/settings/teams.tsx [hunk 21]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{session: Session | null}> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
+      return { redirect: { permanent: false, destination: '/auth/login' } };
+    }
+
+    return {
+      props: { session }
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:179]: Codacy has a fix for the issue: [Replace `······props:·{·session·}` with `····props:·{·session·},`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    props: { session },
```

File: pages/settings/teams.tsx [hunk 22]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
   );
-}
\ No newline at end of file
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{session: Session | null}> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
+      return { redirect: { permanent: false, destination: '/auth/login' } };
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:175]: Codacy has a fix for the issue: [Replace `······return·{·redirect:·{·permanent:·false,·destination:·'/auth/login'` with `····return·{·redirect:·{·permanent:·false,·destination:·"/auth/login"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    return { redirect: { permanent: false, destination: "/auth/login" } };
```

File: pages/settings/teams.tsx [hunk 23]
```diff
@@ -1,46 +1,59 @@
...
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
+    .catch(console.log);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:35]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .catch(console.log);
```

File: pages/settings/teams.tsx [hunk 24]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
       </SettingsShell>
     </Shell>
   );
-}
\ No newline at end of file
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{session: Session | null}> = async (context) => {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:172]: Codacy has a fix for the issue: [Replace `session:·Session·|·null` with `·session:·Session·|·null·`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
export const getServerSideProps: GetServerSideProps<{ session: Session | null }> = async (context) => {
```

File: pages/settings/teams.tsx [hunk 25]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+    const session = await getSession(context);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:173]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  const session = await getSession(context);
```

File: pages/settings/teams.tsx [hunk 26]
```diff
@@ -1,46 +1,59 @@
...
+
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:34]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      })
```

File: pages/settings/teams.tsx [hunk 27]
```diff
@@ -1,46 +1,59 @@
...
-
+export default function Teams() {
   const [session, loading] = useSession();
   const [teams, setTeams] = useState([]);
   const [invites, setInvites] = useState([]);
   const [showCreateTeamModal, setShowCreateTeamModal] = useState(false);
 
-  const loadTeams = () => fetch('/api/user/membership').then((res: any) => res.json()).then(
-    (data) => {
+  const handleErrors = async (resp) => {
```
Comment (emrysal) [inline pages/settings/teams.tsx:20]: getServerSideProps can be removed entirely if you add:

```javascript
if (!resp.ok) {
  // if response status === 403
  // useRouter()
  // return router.push('/auth/login');
  }
}
```

But I'll approve, as I don't consider optimizations like this blocking.

**Review by emrysal — APPROVED** [review]

File: pages/settings/teams.tsx [hunk 28]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{ session: Session | null }> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
+      return { redirect: { permanent: false, destination: '/auth/login' } };
+    }
+
+    return {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:178]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  return {
```

File: pages/api/teams.ts (hunk 2 above)
Comment (baileypumfleet) [inline pages/api/teams.ts:36]: Codacy has a fix for the issue: [Replace `'Team·not·found'` with `"Team·not·found"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  res.status(404).json({ message: "Team not found" });
```

File: pages/settings/teams.tsx [hunk 29]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
   );
-}
\ No newline at end of file
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{ session: Session | null }> = async (context) => {
+    const session = await getSession(context);
+    if (!session) {
+      return { redirect: { permanent: false, destination: '/auth/login' } };
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:175]: Codacy has a fix for the issue: [Replace `······return·{·redirect:·{·permanent:·false,·destination:·'/auth/login'` with `····return·{·redirect:·{·permanent:·false,·destination:·"/auth/login"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    return { redirect: { permanent: false, destination: "/auth/login" } };
```

File: pages/settings/teams.tsx [hunk 30]
```diff
@@ -1,46 +1,59 @@
...
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
+    .then((data) => {
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
+    .catch(console.log);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:35]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .catch(console.log);
```

File: pages/settings/teams.tsx [hunk 31]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
       </SettingsShell>
     </Shell>
   );
-}
\ No newline at end of file
+}
+
+// Export the `session` prop to use sessions with Server Side Rendering
+export const getServerSideProps: GetServerSideProps<{ session: Session | null }> = async (context) => {
+    const session = await getSession(context);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:173]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  const session = await getSession(context);
```

File: pages/settings/teams.tsx [hunk 32]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+    if (!session) {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:174]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  if (!session) {
```

File: pages/settings/teams.tsx [hunk 33]
```diff
@@ -153,4 +166,16 @@ export default function Teams(props) {
...
+      return { redirect: { permanent: false, destination: '/auth/login' } };
+    }
+
+    return {
+      props: { session }
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:179]: Codacy has a fix for the issue: [Replace `······props:·{·session·}` with `····props:·{·session·},`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    props: { session },
```

File: pages/settings/teams.tsx [hunk 34]
```diff
@@ -1,46 +1,59 @@
...
+    if (!resp.ok) {
+      const err = await resp.json();
+      throw new Error(err.message);
+    }
+    return resp.json();
+  }
+
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:30]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then(handleErrors)
```

File: pages/settings/teams.tsx [hunk 35]
```diff
@@ -1,46 +1,59 @@
...
-    }
-  );
+    })
+    .catch(console.log);
+  }
 
-  useEffect(() => { loadTeams(); }, []);
+  useEffect(() => {
+    loadData();
+  }, []);
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:40]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'loadData'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  }, [loadData]);
```

File: pages/api/teams.ts [hunk 36]
```diff
@@ -30,8 +30,8 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       }
     });
 
-    return res.status(201).setHeader('Location', process.env.BASE_URL + '/api/teams/1').send(null);
+    return res.status(201).json({ message: 'Team created' });
```
Comment (baileypumfleet) [inline pages/api/teams.ts:33]: Codacy has a fix for the issue: [Replace `'Team·created'` with `"Team·created"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    return res.status(201).json({ message: "Team created" });
```

File: pages/settings/teams.tsx [hunk 37]
```diff
@@ -1,46 +1,59 @@
...
+      const err = await resp.json();
+      throw new Error(err.message);
+    }
+    return resp.json();
+  }
+
+  const loadData = () => {
+    fetch("/api/user/membership")
+    .then(handleErrors)
+    .then((data) => {
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:31]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then((data) => {
```

File: pages/settings/teams.tsx (hunk 17 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:109]: Codacy has a fix for the issue: [Replace `{invites.map((team)·=>·<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>` with `··{invites.map((team)·=>·(⏎······················<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>⏎····················)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
                    {invites.map((team) => (
```

File: pages/settings/teams.tsx [hunk 38]
```diff
@@ -1,46 +1,59 @@
...
       setTeams(data.membership.filter((m) => m.role !== "INVITEE"));
       setInvites(data.membership.filter((m) => m.role === "INVITEE"));
-    }
-  );
+    })
```
Comment (baileypumfleet) [inline pages/settings/teams.tsx:34]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      })
```

File: pages/settings/teams.tsx (hunk 32 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:174]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  if (!session) {
```

File: pages/settings/teams.tsx (hunk 33 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:179]: Codacy has a fix for the issue: [Replace `······props:·{·session·}` with `····props:·{·session·},`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    props: { session },
```

File: pages/settings/teams.tsx (hunk 34 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:30]: Codacy has a fix for the issue: [Replace `····` with `······`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then(handleErrors)
```

File: pages/settings/teams.tsx (hunk 29 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:175]: Codacy has a fix for the issue: [Replace `······return·{·redirect:·{·permanent:·false,·destination:·'/auth/login'` with `····return·{·redirect:·{·permanent:·false,·destination:·"/auth/login"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
    return { redirect: { permanent: false, destination: "/auth/login" } };
```

File: pages/settings/teams.tsx (hunk 30 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:35]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .catch(console.log);
```

File: pages/settings/teams.tsx (hunk 17 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:109]: Codacy has a fix for the issue: [Replace `{invites.map((team)·=>·<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>` with `··{invites.map((team)·=>·(⏎······················<TeamListItem·onChange={loadData}·key={team.id}·team={team}></TeamListItem>⏎····················)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
                    {invites.map((team) => (
```

File: pages/settings/teams.tsx (hunk 31 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:173]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  const session = await getSession(context);
```

File: pages/api/teams.ts (hunk 2 above)
Comment (baileypumfleet) [inline pages/api/teams.ts:36]: Codacy has a fix for the issue: [Replace `'Team·not·found'` with `"Team·not·found"`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  res.status(404).json({ message: "Team not found" });
```

File: pages/settings/teams.tsx (hunk 37 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:31]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      .then((data) => {
```

File: pages/settings/teams.tsx (hunk 35 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:40]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'loadData'. Either include it or remove the dependency array.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  }, [loadData]);
```

File: pages/settings/teams.tsx (hunk 38 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:34]: Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
      })
```

File: pages/settings/teams.tsx (hunk 28 above)
Comment (baileypumfleet) [inline pages/settings/teams.tsx:178]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689300)
```suggestion
  return {
```

**Pull request merged**